### ISSUE TITLE
Model.create to support "attributes".

### ISSUE THREAD

**Issue by tiblu** [sequelize/sequelize#2402]
**In short**
Have same functionality for **.create** as for **.find** where "attributes" property in the "options" contains the fields result will contain.

Example:

```
 User.create(
            {
                email: 'monkey@banana.com',
                password: getHash('supasecretpasswordforthemonkey'),
            },
            {
                attributes: ['id', 'email', 'createdAt'] // Fields that result will contain, we do not want password, verificationCode etc..
            }
        }).success(function (user) {
             ...
            // result will only contain 'id', 'email', 'createdAt'
            return res.ok(user.toJSON());
        })
```

**The need**

Everywhere I create User (or any object) which contains data that should never be sent in response I have to `delete` that property from the `User.toJSON()` result. Every time a new such property is added to User model, I have to add another `delete` statement.
So in conclusion with this approach accident where sensitive User data is exposed is really easy to happen.

**Other solutions I considered to the problem**
- Separate model (`User.hasOne(SensitiveData`), but that just makes User creation code very difficult to read and maintain.
- User.toJSON() to take "attributes" as input?
